packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    async afterTest (suiteTitle: string | undefined, test: Frameworks.Test) {
+        if (this._framework !== 'mocha' || !this._browser) {
+            return
+        }
+
+        if (!this.shouldRunTestHooks(this._browser, this._accessibility)) {
+            return
+        }
+
+        try {
+            const testIdentifier = this.getIdentifier(test)
+            const accessibilityScanStarted = this._testMetadata[testIdentifier]?.accessibilityScanStarted
+            const shouldScanTestForAccessibility = this._testMetadata[testIdentifier]?.scanTestForAccessibility
+
+            if (!accessibilityScanStarted) {
+                return
+            }
+
+            if (shouldScanTestForAccessibility) {
+                log.info('Automate test case execution has ended. Processing for accessibility testing is underway. ')
+            }
+
+            const dataForExtension = {
+                saveResults: shouldScanTestForAccessibility,
+                testDetails: {
+                    'name': test.title,
+                    'testRunId': process.env.BS_A11Y_TEST_RUN_ID,
+                    'filePath': this._suiteFile,
+                    'scopeList': [suiteTitle, test.title]
+                },
+                platform: this._platformA11yMeta
+            }
+
+            await this.sendTestStopEvent((this._browser as WebdriverIO.Browser), dataForExtension)
+
+            if (shouldScanTestForAccessibility) {
+                log.info('Accessibility testing for this test case has ended.')
+            }
+        } catch (error) {
+            log.error(`Accessibility results could not be processed for the test case ${test.title}. Error :`, error)
+        }
+    }
+
+    /**
+      * Cucumber Only
+    */
+
+    async beforeScenario (world: ITestCaseHookParameter) {
+        if (!this._browser) {
+            return
+        }
+
+        if (!this.shouldRunTestHooks(this._browser, this._accessibility)) {
+            return
+        }
+
+        let pageOpen = true
+        const pickleData = world.pickle
+        const gherkinDocument = world.gherkinDocument
+        const featureData = gherkinDocument.feature
+        const uniqueId = getUniqueIdentifierForCucumber(world)
+        const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions)
+
+        try {
+            this._testMetadata[uniqueId] = {
+                scanTestForAccessibility : shouldScanScenario,
+                accessibilityScanStarted : true
+            }
+            const currentURL = await (this._browser as WebdriverIO.Browser).getUrl()
+            const url = new URL(currentURL)
+            pageOpen = url?.protocol === 'http:' || url?.protocol === 'https:'
+        } catch (e) {
+            pageOpen = false
+        }
+
+        if (!pageOpen) {
+            return
+        }
```

Comment:
I see the same logic in `beforeTest`, maybe we can move this into a utility function?